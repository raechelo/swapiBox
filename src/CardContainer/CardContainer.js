import React, { Component } from 'react';
import Crawl from 'react-star-wars-crawl';
import "react-star-wars-crawl/lib/index.css";
import Loader from '../Loader/Loader';
import People from '../People/People'

class CardContainer extends Component {
  constructor( props ) {
    super( props );
  }

  render() {
    switch(this.props.rendered.currentChoice) {
      case 'people':
      return (
      <section className="Card-Container">
        {this.props.rendered.people.map(p => {
          return (
              <People name={p.name} />
              )
            })
          }
      </section>
      )
      case 'planets':
      break;
      case 'vehicles':
      break;
      case 'crawl':
        return (<Crawl 
          text={this.props.rendered.movie.opening_crawl}
          subtitle={this.props.rendered.movie.release_date}
        />)
      default:
        return <Loader />
    }
  }
}

export default CardContainer;