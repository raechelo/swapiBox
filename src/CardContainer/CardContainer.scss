.Card-Container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  height: 90vh;
  overflow: scroll;
}

.fave {
  font-size: 3rem;
}

.page-btn {
  background-color: hsla(51, 100%, 50%, 0.28);
  border: 3px solid $accent;
  color: $accent;
  font-family: $fancyfont;
  font-size: 1.5rem;
  grid-area: btns;;
  display: block;
  height: 10%;
  margin: auto 0;
  padding: 5px;
  text-transform: uppercase;
  width: 60%;
  z-index: 15;

  &:hover, &:focus {
    background-color: $accent;
    color: black;
    cursor: pointer;
  }
}

.NoMatch {
  z-index: 15;
}

.link, .link:visited {
  text-decoration: none;
}

.back-btn {
  background-color: hsla(51, 100%, 50%, 0.28);
  border: 3px solid $accent;
  color: $accent;
  font-family: $fancyfont;
  font-size: 1.5rem;
  grid-area: btns;;
  display: block;
  height: 10%;
  margin: auto;
  padding: 5px;
  text-transform: uppercase;
  width: 60%;

  &:hover, &:focus {
    background-color: $accent;
    color: black;
    cursor: pointer;
    text-decoration: none;
  }
}