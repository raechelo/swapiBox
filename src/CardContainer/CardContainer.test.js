import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer', () => {

  let rendered = {
    isLoading: true,
    movie: {},
    people: [],
    planets: [], 
    vehicles: [],
    favorites: [],
    currentChoice: '',
  }

  let wrapper;

  beforeEach(()=> {
    wrapper = shallow(
      <CardContainer rendered={rendered} />
    )
  })

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
})