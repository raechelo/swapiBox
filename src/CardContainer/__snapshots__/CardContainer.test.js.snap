// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all the data passed in 1`] = `<Loader />`;
